backend/app/src/user/user.service.ts

```diff
@@ -153,8 +155,21 @@ export class UserService {
     });
   }
 
-  logout(res: Response) {
-    return res.clearCookie('jwtToken', { httpOnly: true });
+  async logout(res: Response, userId: string) {
+    try {
+      await this.prismaService.user.update({
+        where: {
+          id: userId,
+        },
+        data: {
+          status: UserStatus.OFFLINE,
+        },
+      });
+      return res.status(200).clearCookie('jwtToken', { httpOnly: true }).send();
+    } catch (error) {
+      //UserId was not found
+      return res.status(401).send();
```

Comment:
Are we sure this is the right way to handle this error? shouldn't we clear the cookie regardless of errors  in the db? and is a 401 the right error status to send back?

Reply:
No. But are we sure of ? Will the sun rise and shine tomorrow ? ..